doc: Update known issues for 1.9 release (#6611)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

Zigbee
======

.. rst-class:: v1-9-0 v1-8-0

KRKNWK-11720: NCP communication gets stuck
  The communication between the SoC and the NCP Host sometimes stops on the SoC side.
  The device neither sends nor accepts incoming packets.
  Currently, there is no workaround for this issue.

> .. rst-class:: v1-9-0 v1-8-0

Reviewer: This issue is fixed in NCS v1.9
FYI: @maciejpietras

Author: Fixed

Zigbee
======

.. rst-class:: v1-9-0 v1-8-0

KRKNWK-11720: NCP communication gets stuck

>   The communication between the SoC and the NCP Host sometimes stops on the SoC side.

Reviewer 2: This one is still valid, but this list has a different JIRA number for it... https://projecttools.nordicsemi.no/confluence/display/KRKNWK/Release+notes+and+known+issues+1.9.0

Not sure what the status is. @maciejpietras ?

Reviewer 3: 11720 was closed as duplication and is continued in 11704

Author: Updated the JIRA number for this issue to 11704.

KRKNWK-8211: Leave signal generated twice
  The ``ZB_ZDO_SIGNAL_LEAVE`` signal is generated twice during Zigbee Coordinator factory reset.

.. rst-class:: v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

KRKNWK-9714: Device association fails if the Request Key packet is retransmitted

>   If the Request Key packet for the TCLK is retransmitted and the coordinator sends two new keys that are different, a joiner logic error happens that leads to unsuccessful key verification.

Reviewer 2: This one is missing from the following list: https://projecttools.nordicsemi.no/confluence/display/KRKNWK/Release+notes+and+known+issues+1.9.0

Which suggests it might've been fixed. @maciejpietras to confirm.

Reviewer 3: Fixed

Author: Removed the 1-9-0 tag.

  **Workaround:** Call the ``bdb_reset_via_local_action`` function twice to remove all the network information.

.. rst-class:: v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0

KRKNWK-7723: OTA upgrade process restarting after client reset

>   After the reset of OTA Upgrade Client, the client will start the OTA upgrade process from the beginning instead of continuing the previous process.

Reviewer 2: This one is apparently not valid for all releases from 1.6.1 to 1.8.0, and then it suddenly re-appears in 1.9.0, as per this page: https://projecttools.nordicsemi.no/confluence/display/KRKNWK/Release+notes+and+known+issues+1.9.0

@maciejpietras to confirm.

Reviewer 2: OK, so the Confluence page was wrong.

Matter (Project CHIP)
=====================

> 

Reviewer 4: Could we add the following here or should I create a separate PR?
```suggestion

.. rst-class:: v1-9-0

KRKNWK-12950: Android CHIPTool opens the commissioning window using an incorrect PIN code
  Android CHIPTool uses a random instead of a user-provided PIN code to open the commissioning window on a Matter device.

```

HomeKit
=======

.. rst-class:: v1-9-0

> KRKNWK-12842: DFU update from Bluetooth LE only to Thread does not work on nRF5340

Author: Removed the issue.

  DFU will fail when using external flash memory for update image storage.
  For this reason, DFU with external flash memory cannot be performed on HomeKit accessories.

.. rst-class:: v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

KRKNWK-9422: On-mesh commissioning does not work
  Thread's on-mesh commissioning does not work for the HomeKit accessories.

> 

Author: Added this issue.

  **Workaround:** Directed advertising is disabled by default for nRF Desktop.

> .. rst-class:: v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

Author: Removed 1-9-0

Bluetooth® LE
=============

.. rst-class:: v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0

NCSDK-13459: Uninitialized size in hids_boot_kb_outp_report_read
  When reading from the boot keyboard output report characteristic, the :ref:`hids_readme` calls the registered callback with uninitialized report size.

  **Workaround:** Manually cherry-pick and apply commit with fix from main (commit hash: ``f18250dad6cbd9778de7af4b8a774b58e55fe720``).

> .. rst-class:: v1-9-0 v1-8-0

Reviewer 5: Fixed, please remove the v1.9.0 tag.

Common Application Framework (CAF)
==================================

> .. rst-class:: v1-9-0 v1-8-0

Reviewer 6: Fixed in 1.9

Author: Removed 1-9-0

NCSDK-13247: Sensor manager dereferences NULL pointer on wake up for sensors without trigger
  :ref:`caf_sensor_manager` dereferences NULL pointer while handling a :c:struct:`wake_up_event` if a configured sensor does not use trigger.
  This leads to undefined behavior.

  **Workaround** Manually cherry-pick and apply commit with fix from main (commit hash: ``3db6da76206d379c223afe2de646218e60e4f339``).

> .. rst-class:: v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

Author: Removed 1-9-0

Bluetooth® LE
=============

> .. rst-class:: v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0

Reviewer 6: Fixed in 1.9

Author: Removed 1-9-0

Crypto
======

.. rst-class:: v1-9-0

NCSDK-13825: Mbed TLS legacy APIs from Oberon has limited TLS/DTLS support
  The legacy Mbed TLS APIs in Oberon for TLS/DTLS do not support the RSA and DHE algorithms.

.. rst-class:: v1-9-0

> NCSDK-13841: Limited support for RSA in PSA APIs

Reviewer 7: Change to PSA Crypto APIs

Author: Changed for all instances.

  The provided RSA implementation in the PSA APIs has limited support in accelerators.

>   Only the CryptoCell accelerator supports RSA in PSA APIs. Currently, the only supported mode is PKCS1-v1.5. The key size needs to be smaller than 2048 bits and the supported hash functions are SHA-1/SHA-224/SHA-256.

Reviewer 7: Change to PSA Crypto APIs

.. rst-class:: v1-9-0

NCSDK-13843: Limited support for MAC in PSA APIs
  The provided message authentication codes (MAC) implementation in the PSA APIs has limited support in accelerators. Only the CryptoCell accelerator supports MAC operations in PSA APIs and the supported hash algorithms are SHA-1/SHA-224/SHA-256.

.. rst-class:: v1-9-0

NCSDK-13843: Limited support for key derivation in PSA APIs
  The provided key derivation implementation in the PSA APIs has limited support in accelerators. Only the CryptoCell accelerator supports key derivation in PSA APIs and the supported hash algorithms are the SHA-1/SHA-224/SHA-256.

.. rst-class:: v1-9-0

NCSDK-13844: Limited support for GCM in PSA APIs in Oberon
  The provided GCM implementation of the PSA APIs in the Oberon accelerator only supports 12 bytes IV.

.. rst-class:: v1-9-0

> NCSDK-13842: Limited ECC support in PSA APIs

Reviewer 7: ... Towards here (didn't want to list everyone)

Change to PSA Crypto APIs

.. rst-class:: v1-9-0

NCSDK-13843: Limited support for MAC in PSA APIs
  The provided message authentication codes (MAC) implementation in the PSA APIs has limited support in accelerators. Only the CryptoCell accelerator supports MAC operations in PSA APIs and the supported hash algorithms are SHA-1/SHA-224/SHA-256.

.. rst-class:: v1-9-0

NCSDK-13843: Limited support for key derivation in PSA APIs
  The provided key derivation implementation in the PSA APIs has limited support in accelerators. Only the CryptoCell accelerator supports key derivation in PSA APIs and the supported hash algorithms are the SHA-1/SHA-224/SHA-256.

.. rst-class:: v1-9-0

NCSDK-13844: Limited support for GCM in PSA APIs in Oberon
  The provided GCM implementation of the PSA APIs in the Oberon accelerator only supports 12 bytes IV.

.. rst-class:: v1-9-0

NCSDK-13842: Limited ECC support in PSA APIs
  The provided ECDSA implementation in the CryptoCell accelerator does not support 521 bit curves.

>   The provided ECDSA implementation in Oberon only supports the curves secp224r1 and secp256r1.

Reviewer 7: Remove this line (on oberon only supporting curve secp224r1 and secp256r1), this is not a "known issue". This is a limitation of feature which is described in details in configuration...

Author: Removed this.

.. rst-class:: v1-9-0

NCSDK-13843: Limited support for MAC in PSA Crypto APIs
  The provided message authentication codes (MAC) implementation in the PSA Crypto APIs has limited support in accelerators. Only the CryptoCell accelerator supports MAC operations in PSA Crypto APIs and the supported hash algorithms are SHA-1/SHA-224/SHA-256.

.. rst-class:: v1-9-0

NCSDK-13843: Limited support for key derivation in PSA Crypto APIs
  The provided key derivation implementation in the PSA Crypto APIs has limited support in accelerators. Only the CryptoCell accelerator supports key derivation in PSA Crypto APIs and the supported hash algorithms are the SHA-1/SHA-224/SHA-256.

.. rst-class:: v1-9-0

NCSDK-13844: Limited support for GCM in PSA Crypto APIs in Oberon
  The provided GCM implementation of the PSA Crypto APIs in the Oberon accelerator only supports 12 bytes IV.

.. rst-class:: v1-9-0

NCSDK-13842: Limited ECC support in PSA Crypto APIs
  The provided ECDSA implementation in the CryptoCell accelerator does not support 521 bit curves.

> 

Reviewer 8: ```suggestion
NCSDK-13900: Limited AES CBC PKCS7 support in PSA APIs
       The provided implementation in the CryptoCell accelerator for AES CBC with PKCS7 padding does not support multipart APIs.  
```

Author: Added this new issue

In addition to the known issues listed here, see also :ref:`softdevice_controller_limitations` for permanent limitations.

> .. rst-class:: v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

Reviewer 9: DRGN-16650 is fixed in NCS 1.9

Author: Fixed

DRGN-16394: The peripheral accepts a channel map where all channels are marked as bad
  If the initiator sends a connection request with all channels marked as bad, the peripheral will always listen on data channel 0.

> .. rst-class:: v1-9-0 v1-8-0 v1-7-1 v1-7-0

Reviewer 9: DRGN-16317 is fixed for NCS 1.9

Author: Fixed

DRGN-16341: The SoftDevice Controller may discard LE Extended Advertising Reports
  Extended Advertising Reports with data length of 228 are discarded.

> .. rst-class:: v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0

Reviewer 10: This should not have NCS 1.8.0 and NCS 1.9.0?
https://github.com/nrfconnect/sdk-nrf/pull/5784
@rugeGerritsen

Reviewer 9: @thomasstenersen , the following line is correct. The affected versions are placed above the known issue text. So this refers to DRGN-15903 which is present in both NCS 1.8 and NCS 1.9

Trusted Firmware-M (TF-M)
*************************

> 

Reviewer 11: NCSDK-11195: Configuring the BUILD_WITH_TFM configuration SES project configuration or west -t menuconfig results in build errors.
Workaround: Set CONFIG_BUILD_WITH_TFM=y in project configuration file (prj.conf) or through west command line (west build – -DCONFIG_BUILD_WITH_TFM=y)
Affects: from 1.6

Author: Added with a heading. Please check.

> 

Reviewer 11: NCSDK-13530: TF-M Minimal build has increased in size.
TF-M Minimal build exceeds 32kB due to increased dependencies in the TF-M crypto partition.
Affects: 1.9

Author: Added.

> .. rst-class:: v1-9-0 v1-8-0

Author: Removed 1-9-0.

> 

Reviewer 12: KRKNWK-12777: After soft reset TF-M will sometimes trigger a
FLASHACCERR event and halt execution.
Affects: 1.9

NCSDK-12342: When accessing protected storage a SecureFault will
sometimes trigger and execution will halt.
Affects: 1.8 and 1.9

Reviewer 12: NCSDK-12483: Some debug symbols are sometimes missing. The current workaround is to put this in the app CMakeLists.txt: zephyr_link_libraries(-Wl,--undefined=my_missing_debug_symbol)
Affects: 1.8 and 1.9

Reviewer 12: Hi @mia-ko , review would be appreciated.

Author: @SebastianBoe I have added these. Please check if any modifications are required.

NCSDK-11195: Build errors when enabling CONFIG_BUILD_WITH_TFM option
  Configuring the BUILD_WITH_TFM configuration in SES project configuration or using ``west -t menuconfig`` results in build errors.

  **Workaround:** Set ``CONFIG_BUILD_WITH_TFM=y`` in project configuration file (:file:`prj.conf`) or through west command line (``west build – -DCONFIG_BUILD_WITH_TFM=y``).

.. rst-class:: v1-9-0

NCSDK-13530: TF-M minimal build has increased in size
  TF-M minimal build exceeds 32 kB due to increased dependencies in the TF-M crypto partition.

.. rst-class:: v1-9-0

KRKNWK-12777: FLASHACCER event triggered after soft reset
  After soft reset, TF-M sometimes triggers a FLASHACCERR event and execution halts.

.. rst-class:: v1-9-0 v1-8-0

NCSDK-12342: SecureFault exception while accessing protected storage
  When accessing protected storage, a SecureFault exception is sometimes triggered and execution halts.

.. rst-class:: v1-9-0 v1-8-0

NCSDK-12483: Missing debug symbols
  Some debug symbols are missing sometimes.

  **Workaround:** Add the following text in the application :file:`CMakeLists.txt` file: ``zephyr_link_libraries(-Wl,--undefined=my_missing_debug_symbol)``.

> .. rst-class:: v1-8-0
> 
> NCSDK-10022: Missing AES-CFB support in Nordic Security Module

Reviewer 11: We would like to update this known issue, which was not concise enough.

```suggestion
.. rst-class:: v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

NCSDK-10022: TF-M Regression tests fails on AES-CFB tests.
TF-M Regression tests fails on AES-CFB tests because AES-CFB is not supported in :ref:`nrfxlib:nrf_security` (nrf_security).
```

Remove the rest of the description.

Author: Updated

Reviewer 11: Sorry for the mess, there is a bookkeeping error.
This does not affect any releases, and therefore should not be in the known-issues.
Please remove this entry.

Author: No problem! Removed the known issue.

Trusted Firmware-M (TF-M)
*************************

.. rst-class:: v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

NCSDK-11195: Build errors when enabling CONFIG_BUILD_WITH_TFM option
  Configuring the BUILD_WITH_TFM configuration in SES project configuration or using ``west -t menuconfig`` results in build errors.

  **Workaround:** Set ``CONFIG_BUILD_WITH_TFM=y`` in project configuration file (:file:`prj.conf`) or through west command line (``west build – -DCONFIG_BUILD_WITH_TFM=y``).

.. rst-class:: v1-9-0

NCSDK-13530: TF-M minimal build has increased in size

>   TF-M minimal build exceeds 32kB due to increased dependencies in the TF-M crypto partition.

Reviewer 13: ```suggestion
  TF-M minimal build exceeds 32 kB due to increased dependencies in the TF-M crypto partition.
```

Author: Fixed

>   TF-M minimal build exceeds 32KB due to increased dependencies in the TF-M crypto partition.

Reviewer 13: 32KB -> 32 kB

Author: fixed

.. rst-class:: v1-9-0

> KRKNWK-12777: FLASHACCER event after soft reset

Reviewer 13: ```suggestion
KRKNWK-12777: FLASHACCER event triggered after soft reset
```

Author: Added

  After soft reset, TF-M sometimes triggers a FLASHACCERR event and execution halts.

.. rst-class:: v1-9-0 v1-8-0

> NCSDK-12342: SecureFault while accessing protected storage
>   When accessing protected storage a SecureFault is triggered sometimes and execution halts.

Reviewer 13: ```suggestion
NCSDK-12342: SecureFault exception while accessing protected storage
  When accessing protected storage, a SecureFault exception is sometimes triggered and execution halts.
```

Author: Fixed

.. rst-class:: v1-9-0 v1-8-0

NCSDK-12483: Missing debug symbols
  Some debug symbols are missing sometimes.

>   **Workaround:** Add the following text in the app :file:`CMakeLists.txt`: ``zephyr_link_libraries(-Wl,--undefined=my_missing_debug_symbol)``.

Reviewer 13: ```suggestion
  **Workaround:** Add the following text in the application :file:`CMakeLists.txt` file: ``zephyr_link_libraries(-Wl,--undefined=my_missing_debug_symbol)``.
```

Author: fixed
